cpp/oneapi/dal/algo/subgraph_isomorphism/backend/cpu/inner_alloc.hpp

```diff
@@ -18,19 +18,16 @@
 
 #include "oneapi/dal/common.hpp"
 #include "oneapi/dal/detail/common.hpp"
-#include "oneapi/dal/algo/subgraph_isomorphism/detail/byte_alloc.hpp"
-
-namespace oneapi::dal::preview::subgraph_isomorphism::detail {
-struct byte_alloc_iface;
-} // namespace oneapi::dal::preview::subgraph_isomorphism::detail
+#include "oneapi/dal/detail/memory.hpp"
 
 namespace oneapi::dal::preview::subgraph_isomorphism::backend {
-struct inner_alloc {
-    using byte_t = char;
 
-    inner_alloc(detail::byte_alloc_iface* byte_allocator) : byte_allocator_(byte_allocator) {}
-    inner_alloc(const detail::byte_alloc_iface* byte_allocator)
-            : byte_allocator_(const_cast<detail::byte_alloc_iface*>(byte_allocator)) {}
+using byte_alloc_iface_t = oneapi::dal::preview::detail::byte_alloc_iface;
+
+struct inner_alloc {
+    inner_alloc(byte_alloc_iface_t* byte_allocator) : byte_allocator_(byte_allocator) {}
+    inner_alloc(const byte_alloc_iface_t* byte_allocator)
+            : byte_allocator_(const_cast<byte_alloc_iface_t*>(byte_allocator)) {}
```

Comment:
Why the first overload is not enough? 

Suggest to remove the second overload, you secretly casts allocator to non-const, this is not valid. 

`const_cast` is allowed in the very very special cases, it's not the one.

Reply:
thank for comment, if will be fixed with separate PR